backend/infrahub/core/schema/schema_branch.py

```diff
@@ -826,15 +822,61 @@ def validate_default_values(self) -> None:
                         f"{node_schema.namespace}{node_schema.name}: default value {exc.message}"
                     ) from exc
 
+    @staticmethod
+    def _get_all_uc_not_overlapping(uniqueness_constraints: list[list[str]]) -> list[set[str]]:
+        """
+        Compute all combinations of uniqueness constraints from a given set of constraints.
+        For instance,
+        Input [["name"], ["age"], ["color", "salary"]] would return
+
+        [{'name'}, {'age'}, {'color', 'salary'}, {'age', 'name'}, {'name', 'color', 'salary'},
+         {'age', 'color', 'salary'}, {'age', 'name', 'color', 'salary'}]
+        """
+
```

Comment:
Note this greedy algorithm generates 2^n constraints for n input constraints. I don't expect of that to be an issue as number of uniqueness constraints is usually very small, but maybe we could restrict number of uniqueness constraints to avoid memory errors (even though, there would probably be no realistic use case for that), or potentially use a smarter algorithm if someone has a suggestion.